dagfactory/dagbuilder.py

```diff
@@ -726,6 +726,85 @@ def process_file_with_datasets(file: str, datasets_conditions: str) -> Any:
             datasets_uri = utils.get_datasets_uri_yaml_file(file, list(dataset_map.keys()))
             return [Dataset(uri) for uri in datasets_uri]
 
+    @staticmethod
+    def _asset_schedule(value):
```

Comment:
This function is quite complex. Would it be possible to break it down into smaller functions, that could be individually tested and changed having a smaller impact?